flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/proxy/KinesisProxyV2Factory.java

```diff
@@ -47,22 +50,28 @@
     public static KinesisProxyV2Interface createKinesisProxyV2(final Properties configProps) {
         Preconditions.checkNotNull(configProps);
 
-        final ClientConfiguration clientConfiguration =
-                new ClientConfigurationFactory().getConfig();
+        final AttributeMap convertedProperties = AwsV2Util.convertProperties(configProps);
+        final AttributeMap.Builder clientConfiguration = AttributeMap.builder();
         populateDefaultValues(clientConfiguration);
 
         final SdkAsyncHttpClient httpClient =
-                AwsV2Util.createHttpClient(
-                        clientConfiguration, NettyNioAsyncHttpClient.builder(), configProps);
+                AWSGeneralUtil.createAsyncHttpClient(
+                        convertedProperties.merge(clientConfiguration.build()),
+                        NettyNioAsyncHttpClient.builder());
         final FanOutRecordPublisherConfiguration configuration =
                 new FanOutRecordPublisherConfiguration(configProps, emptyList());
+
+        Properties legacyConfigProps = new Properties(configProps);
+        legacyConfigProps.setProperty(
+                AWSKinesisDataStreamsConfigConstants.LEGACY_CONNECTOR, Boolean.toString(true));
```

Comment:
Maybe instead of setting a boolean here, we can just set the user agent header? This way we can remove this logic from the new code